template.yaml

```diff
@@ -297,42 +297,52 @@ Resources:
         PasswordLength: 24
         ExcludePunctuation: true
       KmsKeyId: !Ref SecretsKmsKey
-      Tags: 
+      Tags:
         - Key: !Sub ${TagNameParameter}
-          Value: !Sub ${TagValueParameter}      
+          Value: !Sub ${TagValueParameter}
 
   # Attach Canvas database user Secret to the cluster
   DatabaseUserSecretAttachmentCanvas:
     Type: AWS::SecretsManager::SecretTargetAttachment
     Properties:
       SecretId: !Ref DatabaseUserSecretCanvas
-      TargetId: !If [ExistingDatabase, !Select [ "6", !Split [ ":" , !Ref DatabaseClusterArnParameter ] ], !Ref AuroraDatabaseCluster]      
-      TargetType: AWS::RDS::DBCluster          
+      TargetId: !If [ExistingDatabase, !Select [ "6", !Split [ ":" , !Ref DatabaseClusterArnParameter ] ], !Ref AuroraDatabaseCluster]
+      TargetType: AWS::RDS::DBCluster
 
   DatabaseSubnetGroup:
     Type: AWS::RDS::DBSubnetGroup
     Properties:
       DBSubnetGroupDescription: !Sub ${ResourcePrefixParameter}-cd2-subnetgroup-${EnvironmentParameter}
       SubnetIds: !Ref DatabaseSubnetListParameter
-      Tags: 
+      Tags:
         - Key: !Sub ${TagNameParameter}
-          Value: !Sub ${TagValueParameter}      
+          Value: !Sub ${TagValueParameter}
 
   DatabaseClientSecurityGroup:
     Type: AWS::EC2::SecurityGroup
     Properties:
       GroupDescription: !Sub ${ResourcePrefixParameter}-cd2-database-client-sg-${EnvironmentParameter}
       VpcId: !Ref VpcIdParameter
-      Tags: 
+      SecurityGroupEgress:
+        - IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+          CidrIp: 0.0.0.0/0
+      Tags:
         - Key: !Sub ${TagNameParameter}
-          Value: !Sub ${TagValueParameter}      
+          Value: !Sub ${TagValueParameter}
 
   EmptySecurityGroup:
     Type: AWS::EC2::SecurityGroup
     Properties:
       GroupDescription: !Sub ${ResourcePrefixParameter}-cd2-empty-sg-${EnvironmentParameter}
       VpcId: !Ref VpcIdParameter
-      Tags: 
+      SecurityGroupEgress:
+        - IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+          CidrIp: 0.0.0.0/0
+      Tags:
         - Key: !Sub ${TagNameParameter}
           Value: !Sub ${TagValueParameter}
```

Comment:
It's not specific to your change, but the naming of this resource has always bugged me. `ListFunctionSecurityGroup` is more descriptive than `EmptySecurityGroup`. Can you change it and also add a `GroupName` like `Canvas Data 2 ListFunction`?